### DISCUSSION TITLE
Example of how to only check/format staged files?

### DISCUSSION THREAD

**Discussion by tethridge** [cortinico/ktfmt-gradle#335] [Q&A] [answered]
I've read through the documentation under the pre-commit hook section and I've attempted to get a check/format pre-commit script to work, but have not been successful.  Can somebody provide more specific steps that explain how to configure the `--include-only` option.  I have the pre-commit check successfully checking the entire project on each commit.  I'm hoping to reduce the time of the task.

**Comment by simonhauck** [comment]
To my knowledge, includeonly is just a list of files that should be formatted.

There is no option to to apply the formatting only to changed/staged files. Is that what you are trying to achieve?

**Comment by cortinico** [comment]
You will have to ask `git` which files have been modified and pass them to the `--include-only` parameter. You can do so inside a bash script

**Comment by tethridge** [comment, marked as answer]
This is the pre-commit script that I have pulled together.  It doesn't work because "--include-only" doesn't work like I have it.  I have the list of files delimited by ':', but I get an error that the "--include-only" option doesn't exist.  

If I remove "--include-only=$CHANGED_FILES" from the two lines that invoke ktfmt, then it works.  However, it will check the whole project which can be slow.

```
#!/usr/bin/env bash
echo "Running Kotlin formatting check..."
OUTPUT="/tmp/ktfmt-$(date +%s)"
CHANGED_FILES=`git diff --name-only --cached --diff-filter=AMd | paste -sd ":" -`
./gradlew ktfmtCheck --include-only=$CHANGED_FILES  > $OUTPUT 2>&1
EXIT_CODE=$?
if [ $EXIT_CODE -ne 0 ]; then
  ./gradlew ktfmtFormat --include-only=$CHANGED_FILES > $OUTPUT 2>&1
  #cat $OUTPUT
  rm $OUTPUT
  echo "**************************************************************************"
  echo "* FOUND IMPROPERLY FORMATTED FILES - ktfmtFormat was run.                *"
  echo "* Check results for accuracy, add changed files, and commit again.       *"
  echo "**************************************************************************"
  git status
  exit $EXIT_CODE
fi
rm $OUTPUT
```

**Reply by cortinico** [reply]
The include only accepts both `,` and `:` for file separator:
https://github.com/cortinico/ktfmt-gradle/blob/e4b89f07700f601fdad3caf21fc7186c6d50812d/plugin-build/plugin/src/main/java/com/ncorti/ktfmt/gradle/tasks/IncludedFilesParser.kt#L9

The problem you're having is that you need a custom task as described here:
https://github.com/cortinico/ktfmt-gradle#using-with-a-pre-commit-hook-

**Reply by tethridge** [reply]
I think I see the issue that I was missing.  I was trying to run the check on only the files that were staged and then format those staged files if the check failed.  I see that the example is only for formatting.  When I try to create a ktfmtPrecommitCheck and a ktfmtPrecommitFormat tasks that I can apply the include-only flag to limit the scope.  It seems to limit the scope as expected, but the check will always return success.  So, does that mean that you can only use the include-only flag to limit the formatting?

**Reply by cortinico** [reply]
Nope the `include-only` flag is valid for both Check and Format tasks

**Reply by tethridge** [reply]
Can you explain which build.gradle.kts file I should add the custom task and ktfmt block so that this works as expected?  I've tried a lot of combinations and I have gotten some strange results.  My project is a KMP project for Android/iOS that uses Compose Multiplatform.

**Reply by cortinico** [reply]
If you could create a repro using this template: https://github.com/cortinico/kotlin-android-template and applying your changes I could take a look at it

**Reply by tethridge** [reply]
Alright, it seems to work.  Is this the recommended way to configure ktfmt?  https://github.com/tethridge/ktfmtTest/commit/e8baa9edc40457ae168c48565ca4de0ed1037fd5

**Reply by cortinico** [reply]
Yup that seems correct
